### PR TITLE
✨ [source-gainsight-px] add new streams

### PR DESCRIPTION
**Pull request by honggyu420**
## What
Adds many new streams

## Recommended reading order
1. manifest.yaml
2. schemas/*

## 🚨 User Impact 🚨
None, adds new streams to existing connector

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-gainsight-px/source_gainsight_px/manifest.yaml [hunk 1]
```diff
@@ -109,6 +170,65 @@ definitions:
...
+      extractorPath: "segments"
+    type: DeclarativeStream
+    retriever:
+      $ref: "#/definitions/custom_retriever"
+    name: "segments"
+
+  segment_match_events_stream:
+    $parameters:
+      path: "events/segment_match"
+      extractorPath: "featureMatchEvents"
```
Comment (honggyu420) [inline airbyte-integrations/connectors/source-gainsight-px/source_gainsight_px/manifest.yaml:205]: this is not a typo. Actually stated in the docs like this
https://gainsightpx.docs.apiary.io/#reference/segmentmatchevent/v1eventssegmentmatch/get-segment-match-events

**Comment by natikgadzhi** [comment]
Hey @honggyu420, thank you for the contribution!

This connector needs a little bit of maintenance:
1. Let's switch away from custom Docker image. 
2. Let's switch from setup.py to Poetry
3. Upgrade to latest `airbyte_cdk`

It seems that `source-gainsight-px` only uses declarative syntax without any custom Python components, so we could also pass the connector through the Connector Builder UI, that would inline the schemas and reduce the amount of files out here, but it would kill the ref blocks (and duplicate their contents). 

I will try and get through the 1-3 points and record a little demo of how to do it for this and other connectors.

**Comment by natikgadzhi** [comment]
@honggyu420 @honggyu-rr I've rebased the PR and added the commit to move to base image in https://github.com/airbytehq/airbyte/pull/36221 — you can cherry-pick my last commit into this PR and rebase as well. 

If we don't have a sandbox account for gainsight-px (not sure), we might need them to get this merged.

**Comment by natikgadzhi** [comment]
I've also commmitted the Poetry bump — I think it might be easier for me to merge my PR — you will still get the contribution score since I cherry-picked your commits. Good with you, @honggyu-rr?

File: airbyte-integrations/connectors/source-gainsight-px/metadata.yaml [hunk 2]
```diff
@@ -14,7 +14,7 @@ data:
   connectorSubtype: api
   connectorType: source
   definitionId: 0da3b186-8879-4e94-8738-55b48762f1e8
-  dockerImageTag: 0.1.0
+  dockerImageTag: 0.1.1
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-gainsight-px/metadata.yaml:17]: New stream is a minor update. Please update to 0.2.0

File: airbyte-integrations/connectors/source-gainsight-px/integration_tests/configured_catalog.json [hunk 3]
```diff
@@ -67,10 +213,12 @@
...
+        "supported_sync_modes": [
+          "full_refresh"
+        ]
       },
       "sync_mode": "full_refresh",
       "destination_sync_mode": "overwrite"
     }
   ]
-}
+}
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-gainsight-px/integration_tests/configured_catalog.json:224]: EOF

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Please run the `airbyte-ci format all fix` to fix all files.

**Comment by marcosmarxm** [comment]
Closed as inactive.

**Pull request closed without merging**